genie-agent/src/main/java/com/netflix/genie/agent/AgentMetadataImpl.java

```diff
@@ -0,0 +1,86 @@
+/*
+ *
+ *  Copyright 2018 Netflix, Inc.
+ *
+ *     Licensed under the Apache License, Version 2.0 (the "License");
+ *     you may not use this file except in compliance with the License.
+ *     You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *     Unless required by applicable law or agreed to in writing, software
+ *     distributed under the License is distributed on an "AS IS" BASIS,
+ *     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *     See the License for the specific language governing permissions and
+ *     limitations under the License.
+ *
+ */
+
+package com.netflix.genie.agent;
+
+import com.netflix.genie.GenieAgentApplication;
+import lombok.Getter;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.context.annotation.Lazy;
+import org.springframework.stereotype.Component;
+
+import java.lang.management.ManagementFactory;
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Implementation of AgentMetadata.
+ * @author mprimi
+ * @since 4.0.0
+ */
+@Slf4j
+@Component
+@Getter
+@Lazy
+class AgentMetadataImpl implements AgentMetadata {
+
+    private static final String FALLBACK_STRING = "unknown";
+    private final String agentVersion;
+    private final String agentHostName;
+    private final String agentPid;
+
+    AgentMetadataImpl() {
+        this.agentVersion = getAgentVersionOrFallback();
+        this.agentHostName = getAgentHostNameOrFallback();
+        this.agentPid = getAgentPidOrFallback();
+    }
+
+    private static String getAgentVersionOrFallback() {
```

Comment:
Why did you make all these static?

Reply:
No particular reason, they enforces they're pure functions.